### PR TITLE
Add spar-migrate-data executable

### PR DESCRIPTION
**Pull request by smatting**
This PR adds `spar-migrate-data` data migration tool to spar, similar than `galley-migrate-data`.

It doesn't contain any migrations and running it is a no-op.

This PR also converts the `migrate-external-ids ` executable introduced in #1384 to a migration
that can be added to list of migrations in a future PR.

### CODE REVIEW

File: services/spar/src/Spar/DataMigration/V1_ExternalIds.hs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+migrationMain ::
+  ( HasSpar env,
+    HasBrig env,
+    HasLogger env,
+    HasMigEnv env,
+    HasFailCount env
+  ) =>
+  RIO env ()
+migrationMain = do
```
Comment (fisx) [inline services/spar/src/Spar/DataMigration/V1_ExternalIds.hs:129]: I'm assuming that the rest of this module hasn't changed much, and have just glanced through it.

**Review by fisx — APPROVED** [review]
Lots of comments, but I don't want to insist on any of them.  Ship it or change things first as you deem fit!

File: services/spar/src/Spar/DataMigration/Run.hs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+cleanup :: (MonadIO m) => Env -> m ()
+cleanup env = do
+  C.shutdown (sparCassandra env)
+  C.shutdown (brigCassandra env)
+  Log.close (logger env)
+
+runMigrations :: Env -> [Migration] -> IO ()
+runMigrations env migrations = do
+  vmax <- latestMigrationVersion env
+  let pendingMigrations = filter (\m -> version m > vmax) migrations
```
[the remark this answers is not in the document]
Comment (smatting) [reply services/spar/src/Spar/DataMigration/Run.hs]: In my defense I just copied from galley ;)

File: services/spar/src/Spar/DataMigration/V1_ExternalIds.hs (hunk 1 above)
Comment (smatting) [reply services/spar/src/Spar/DataMigration/V1_ExternalIds.hs]: Yes, that module hasn't changed

File: services/spar/src/Spar/DataMigration/Options.hs [hunk 3]
```diff
@@ -38,40 +33,15 @@ import Control.Lens
...
 
 settingsParser :: Parser MigratorSettings
 settingsParser =
   MigratorSettings
     <$> cassandraSettingsParser "spar"
     <*> cassandraSettingsParser "brig"
-    <*> flag NoDebug Debug (long "debug")
-    <*> flag NoDryRun DryRun (long "dry-run" <> short 'n')
+    <*> switch (long "debug")
+    <*> switch (long "dry-run" <> short 'n')
```
[the remark this answers is not in the document]
Comment (smatting) [reply services/spar/src/Spar/DataMigration/Options.hs]: I reverted it

**Pull request merged**